### DISCUSSION TITLE
Using the node API fails when app is bundled using @vercel/ncc

### DISCUSSION THREAD

**Discussion by DanielPe05** [vuepress/core#87] [Q&A] [answered]
Hi friends!!!
I'm running into an issue when bundling my app using @vercel/ncc (I need to use ncc because I'm trying to combine node_modules into 1 file for distribution):

```
Could not resolve entry module (dist/lib/server.js)
Could not resolve entry module (dist/lib/client.js)
```
Do you know if there is something that could be done from the vuepress side to work around this issue?

This is the bundle script:
```
"bundle": "NODE_ENV=production ncc build --source-map --license licenses.txt"
```

Versions:
```
...
"@vercel/ncc": "^0.27.0",
"vuepress": "^2.0.0-beta.6",
"vuepress-vite": "^2.0.0-beta.2"
...
```
For some reason, ncc doesn't seem to create the lib directory after it bundles things up.

I'm running the script to build the docs using the node API. Looks a bit like this:

```javascript
import path from 'path'
import { createApp } from 'vuepress'

const buildDocs = async (space: string) => {
    const source = path.resolve(__dirname, '.temp-docs')
    const app = createApp({
      source,
      title: `${capitalize(space)} Documentation`,
      description: `Documentation for ${capitalize(space)} space`,
      base: '/',
      themeConfig: {
        sidebarDepth: 2,
        sidebar: 'auto'
      }
    })

    await app.init()
    await app.prepare()

    return app.build()
  }
```
Any help here is appreciated.

**Comment by meteorlxy** [comment, marked as answer]
So everything works well without `@vercel/ncc` ?

If so, that could be an issue of `@vercel/ncc`. I'm not familiar with that project, but if you want to bundle your dist files, consider using rollup.
